File: src/util/entities/Stream.ts
import { Column, Entity, JoinColumn, ManyToOne, RelationId } from "typeorm";
import { BaseClass } from "./BaseClass";
import { User } from "./User";
import { Channel } from "./Channel";

@Entity({
    name: "streams",
})
export class Stream extends BaseClass {
    @Column()
    @RelationId((stream: Stream) => stream.owner)
    owner_id: string;

    @JoinColumn({ name: "owner_id" })
    @ManyToOne(() => require("./User").User, {
        onDelete: "CASCADE",
    })
    owner: User;

    @Column()
    @RelationId((stream: Stream) => stream.channel)
    channel_id: string;

    @JoinColumn({ name: "channel_id" })
    @ManyToOne(() => require("./Channel").Channel, {
        onDelete: "CASCADE",
    })
    channel: Channel;

    @Column()
    endpoint: string;
}

File: extra/admin-api/Models/Spacebar.Models.Db/Models/Stream.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("streams")]
public partial class Stream
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("owner_id")]
    public long OwnerId { get; set; }

    [Column("channel_id")]
    public long ChannelId { get; set; }

    [Column("endpoint", TypeName = "character varying")]
    public string Endpoint { get; set; } = null!;

    [ForeignKey("ChannelId")]
    [InverseProperty("Streams")]
    public virtual Channel Channel { get; set; } = null!;

    [ForeignKey("OwnerId")]
    [InverseProperty("Streams")]
    public virtual User Owner { get; set; } = null!;

    [InverseProperty("Stream")]
    public virtual ICollection<StreamSession> StreamSessions { get; set; } = new List<StreamSession>();
}
